packages/ui-components/src/components/Badge/Badge.component.tsx

```diff
@@ -30,27 +60,24 @@ const badgeVariantStyles = {
   error: "jn-bg-theme-error/25",
 }
 
-const getVariantStyle = (variant: VariantType): string => badgeVariantStyles[variant] || badgeVariantStyles.default
-
-// ICON UTILS
+const getVariantStyle = (variant: BadgeVariantType): string => badgeVariantStyles[variant] || badgeVariantStyles.default
 
 const iconStyles = "jn-mr-1 jn-items-center"
 
 // Get icon based on the provided variant or icon string
-const getIcon = (icon: boolean | KnownIcons | undefined, variant: VariantType): KnownIcons | undefined => {
+const getIcon = (icon: boolean | KnownIcons | undefined, variant: BadgeVariantType): KnownIcons | undefined => {
   if (typeof icon === "string" && isValidIcon(icon)) return icon
   if (icon === true) return variant as KnownIcons
```

Comment:
remember my code that I told you to construct the type? We should change this line, I already sent you a snippets in slack yesterday.

Reply:
Thanks, @barsukov. I was waiting for a decision on the critical variant, as [here](https://github.com/cloudoperators/juno/pull/514#discussion_r1800561824). 
It was decided that the `critical` variant shouldn't exist at all, so we can remove it all together and also just remove the forced casting: `variant as KnownIcons`! Wooo. 🚀
I think that we don't need the solution from yesterday? But let me know if I missed something.